src/web/app/pages-instructor/student-list/student-list.component.ts

```diff
@@ -114,4 +168,61 @@ export class StudentListComponent implements OnInit {
   isStudentToHide(studentEmail: string): boolean {
     return this.listOfStudentsToHide.indexOf(studentEmail) > -1;
   }
+
+  /**
+   * Sorts the student list
+   */
+  sortStudentListEvent(by: SortBy): void {
+    this.tableSortBy = by;
+    this.tableSortOrder =
+        this.tableSortOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
+    this.students.sort(this.sortBy(by));
```

Comment:
In addition, there is a problem here: You are changing the data from the parent component.

`StudentListInfoTableComponent` provides a solution to this.

Reply:
The `this.students` is the flattened list and not the input from the parent so the data from the parent should not be affected.